mbed_tools_ci/tag_and_release.py

```diff
@@ -43,51 +44,47 @@ def tag_and_release(mode: CommitType,
         raise ValueError('Undefined version.')
     if mode == CommitType.DEVELOPMENT:
         return
-    _update_repository(mode, is_new_version, version, current_branch)
+    docs_dir = _update_documentation()
+    _update_repository(mode, is_new_version, version, current_branch, docs_dir)
     if is_new_version:
         _release_to_pypi()
 
 
-def _remove_old_documentation(git: GitWrapper, docs_dir: Path) -> None:
-    """Remove existing docs first to make sure no stale pages left behind."""
-    if docs_dir.is_dir():
-        shutil.rmtree(str(docs_dir))
-        git.add(str(docs_dir))
-
-
-def _add_new_documentation(
-        git: GitWrapper, temp_docs_dir: Path, docs_dir: Path) -> None:
-    """Move across documentation from the temporary file and add to git."""
-    shutil.move(str(temp_docs_dir), str(docs_dir))
-    git.add(str(docs_dir))
-
-
-def _update_documentation(git: GitWrapper) -> None:
+def _get_documentation_paths() -> Tuple[Path, Path]:
     docs_dir = Path(configuration.get_value(
         ConfigurationVariable.DOCUMENTATION_PRODUCTION_OUTPUT_PATH
     ))
-    # Pdoc nests documentation in a folder with the module's name.
-    # Only transfer the contents of this folder.
-    temp_docs_dir = Path(
-        configuration.get_value(
-            ConfigurationVariable.DOCUMENTATION_TEMP_CI_OUTPUT_PATH
-        ),
+    docs_contents_dir = docs_dir.joinpath(
         configuration.get_value(
             ConfigurationVariable.MODULE_TO_DOCUMENT
         )
     )
+    return docs_dir, docs_contents_dir
 
-    _remove_old_documentation(git, docs_dir)
-    _add_new_documentation(git, temp_docs_dir, docs_dir)
+
+def _update_documentation() -> Path:
+    """Ensures the documentation is in the correct location for releasing.
+
+    Pdoc nests its docs output in a folder with the module's name.
+    This process removes this unwanted folder.
+    """
+    docs_dir, docs_contents_dir = _get_documentation_paths()
+    with TemporaryDirectory() as temp_dir:
+        shutil.move(str(docs_contents_dir), str(temp_dir))
+        shutil.rmtree(docs_dir)
+        shutil.move(str(temp_dir), str(docs_dir))
+    return docs_dir
 
 
 def _update_repository(mode: CommitType, is_new_version: bool,
```

Comment:
It looks like the build is failing because this has too many arguments, but I have removed any additions I made. Any ideas why it is now not happy?